app/src/main/kotlin/com/strv/archdemo/ktools/Preferences.kt

```diff
@@ -0,0 +1,15 @@
+package com.strv.archdemo.ktools
+
+import android.content.SharedPreferences
+
+private const val PREFS_COUNTER = "prefs_counter"
+
+class Preferences(val prefs: SharedPreferences) {
+    fun getCounterValueFromPreferences(): Int {
+        return prefs.getInt("PREFS_COUNTER", 0)
+    }
+
+    fun updateCounterValueInPreferences(newValue: Int) {
+        prefs.edit().putInt("PREFS_COUNTER", newValue).apply()
```

Comment:
Not sure about what would this codebase serve for, but I'd add some comments to the code just in case it should be used as a study materials. 
E.g. here difference between `.apply()` and `.commit()`  and why the `apply()` is used.